backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -1535,26 +1547,30 @@ private ExperimentRun filteredDatasetsBasedOnPrivileges(
   }
 
   private Map<String, List<KeyValue>> getExperimentRunHyperparameterConfigBlobMap(
-      Session session, List<String> expRunIds, List<String> selfAllowedRepositoryIds) {
+      Session session, List<String> expRunIds, List<Resources> selfAllowedRepositories) {
 
     var queryBuilder =
         "Select vme.experimentRunEntity.id, cb From ConfigBlobEntity cb INNER JOIN VersioningModeldbEntityMapping vme ON vme.blob_hash = cb.blob_hash WHERE cb.hyperparameter_type = :hyperparameterType AND vme.experimentRunEntity.id IN (:expRunIds) ";
 
+    Set<String> accessibleResourceIds;
     if (mdbConfig.isPopulateConnectionsBasedOnPrivileges()) {
-      if (selfAllowedRepositoryIds == null || selfAllowedRepositoryIds.isEmpty()) {
+      boolean allowedAllResources =
+          RoleServiceUtils.checkAllResourceAllowed(selfAllowedRepositories);
+      if (allowedAllResources) {
         return new HashMap<>();
       } else {
+        accessibleResourceIds = RoleServiceUtils.getResourceIds(selfAllowedRepositories);
         queryBuilder = queryBuilder + " AND vme.repository_id IN (:repoIds)";
       }
+    } else {
+      accessibleResourceIds = null;
```

Comment:
null? not an empty set? also, if you initialized the variable to this value, you wouldn't need to do it in an else block.

Reply:
null should be alright here too, but ok, replaced it with empty set and moved to initializer